quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -1291,9 +1296,23 @@ impl MetastoreService for PostgresqlMetastore {
 
     async fn open_shards(
```

Comment:
There might be a way to perform only one pass per index:
```SQL
INSERT INTO shards (index_uid, source_id, leader_id, follower_id, shard_state, publish_position_inclusive, publish_token)
VALUES ($1, $2, $3, $4, $5, $6, $7)
ON CONFLICT (index_uid, source_id, shard_id)
DO NOTHING
RETURNING *;
```

Reply:
Oh okay I have never tried this out before, I will update my code then.